sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -102,6 +102,27 @@ object functions {
     Column(literalExpr)
   }
 
+  /**
+   * Creates a [[Column]] of literal value.
+   *
+   * The passed in object is returned directly if it is already a [[Column]].
+   * If the object is a Scala Symbol, it is converted into a [[Column]] also.
+   * Otherwise, a new [[Column]] is created to represent the literal value.
+   *
+   * @group normal_funcs
+   * @since 2.2.0
+   */
+  def typedLit[T : TypeTag](literal: T): Column = {
+    literal match {
```

Comment:
@hvanhovell How about the fix? https://github.com/apache/spark/pull/16610/commits/4f65fe2315e35272aecbea9e2e3ef1f15398da31